build.py

```diff
@@ -193,6 +193,20 @@ def use_python_env(folder):
             text=True,
             cwd=root_dir,
         )
+
+    step_start("Runing Q# format check")
```

Comment:
- I think this code should be contained inside the same `if` statement above, so that it only runs if we're building any Rust code.
- Instead of `step_start` could you simply do a `print()`? Unfortunately `step_start`/`step_end` logging doesn't support nested pairs (the pert timer gets thrown off) - and this is missing a `step_end` anyways

Reply:
yes - `print()` instead of `step_start()`, but why would we only want to check the format of the Q# code if we compile Rust?

Reply:
> why would we only want to check the format of the Q# code if we compile Rust?

Warning, subjective answer ahead :)

Because if I'm running `./build.py` with specific flags that skip compiling the Rust bits (which I often do, when working on just the JS/Python bits), it would be an annoying extra step to build & run the Q# formatter.

Almost everything in this file is contained within these `if build_xyz` checks and I think it's good to follow that example, instead of running anything unconditionally. The idea is, if you're working on a specific component, you want to be able to build & check only those components. 

So in which cases _should_ we run the formatter? There's no ideal answer but these would be my expectations:

- if I'm building the library (`build_cli`)
- if I'm building the samples (`build_samples`)
- if I'm building the formatter itself (`build_cli` again)

Some common use cases and my expectations on whether the Q# formatter should run:

`./build.py` - yes (since no arguments means ALL)
`./build.py --npm --vscode` - no
`./build.py --samples` - yes
`./build.py --cli` - yes
`./build.py --wasm --npm --vscode`  - no


So maybe I'm changing my answer slightly: `if build_cli or build_samples` seems like the ideal condition to check.

Reply:
The number of possible option combinations is a bit large, so I'll defer to you on this.

Reply:
done, `if build_cli or build_samples`